wallet/src/repository/wallet/mod.rs

```diff
@@ -372,8 +364,24 @@ where
             }
         }
 
+        // Append balance movements of pending blocks
+        state.pending_movements.values().for_each(|x| {
+            keyed_transactions.extend(x.iter().map(|movement| (movement.db_key, movement.clone())));
+        });
+
+        let total_txns = keyed_transactions.len();
+        // Append local pending balance movements (not yet included in blocks)
+        let local_movements = state.local_movements.values().cloned().collect::<Vec<_>>();
+        keyed_transactions.extend(local_movements.iter().enumerate().map(|(index, movement)| {
+            (
+                u32::try_from(total_txns + index)
+                    .expect("Transaction length should be convertible to u32"),
+                movement.clone(),
+            )
+        }));
+
         Ok(model::Transactions {
-            transactions,
+            transactions: keyed_transactions.values().cloned().rev().collect(),
```

Comment:
Here you don't need to clone the elements, you should be able to consume `keyed_transactions` using `into_iter`:

```suggestion
            transactions: keyed_transactions.into_iter().map(|(k, v)| v).rev().collect(),
```

Reply:
The clone wars! :rofl: